lib/connections/mongoose.js

```diff
@@ -3,6 +3,8 @@ import Promise from 'bluebird';
 import logger from 'lib/logger';
 import defaultTo from 'lodash/defaultTo';
 
+/** @typedef {import("./mongoose.types").Connection} Connection */
```

Comment:
Is there any reason we can't do the following instead to avoid having our own file to mirror the Mongoose types?

```
/** @typedef {import('mongoose').Connection} Connection */
```

Reply:
I thought that it will not work because trying to jump to that type shows dropdown with corresponding type selection
<img width="546" alt="Screenshot 2020-01-16 at 18 31 24" src="https://user-images.githubusercontent.com/6347345/72543464-6fc64e00-388e-11ea-855d-3e759c6551b0.png">
<img width="570" alt="Screenshot 2020-01-16 at 18 31 27" src="https://user-images.githubusercontent.com/6347345/72543467-705ee480-388e-11ea-8976-25b91f43d8d7.png">

But it seems it really works:
<img width="460" alt="Screenshot 2020-01-16 at 18 32 50" src="https://user-images.githubusercontent.com/6347345/72543567-9edcbf80-388e-11ea-9d34-fd0b80beda5f.png">

But in this case we have an issue that I described in comment to this PR related to CPU load

Reply:
Ok so you're having performance issues with this?

Reply:
Now everything is ok, but as the solution we need to use different name for `@typedef` type like so:

```
/** @typedef {module:mongoose.Connection} MongooseConnection */
```